app/assets/javascripts/discourse/tests/acceptance/tags-test.js

```diff
@@ -539,6 +539,27 @@ acceptance("Tag info", function (needs) {
     );
   });
 
+  test("admin can edit tag synonyms and create new tags", async function (assert) {
+    updateCurrentUser({ moderator: false, admin: true });
+    await visit("/tag/happy-monkey");
+    await click("#show-tag-info");
+    await click("#edit-synonyms");
+
+    const addSynonymsDropdown = selectKit("#add-synonyms");
+    await addSynonymsDropdown.expand();
+
+    await fillIn(".filter-input", "cheeky-monkey");
+    await click(".select-kit-collection .select-kit-row");
+
+    assert.ok(exists(".add-synonyms .ok"), "can see add synonym button");
+    await click(".add-synonyms .ok");
+
+    assert.ok(
+      exists(".dialog-footer .btn-primary"),
+      "can see confirm synonym button"
+    );
```

Comment:
Does these actions trigger a request to the server? Since this test is testing the ability to edit tag synonyms and create new tags, I think we need to assert that the request sent to the server is correct when saving the changes. Otherwise, we're just testing that the UI works.

Reply:
If we're limiting this acceptance test to just client side, I think a test to assert that the right POST/PUT request to server is important. 

If a system test is added, then we just need to ensure that the new tag is reflected in the UI after being added to the synonym.